src/validation.ts

```diff
@@ -40,7 +40,7 @@ export function generateValidator<T>(
 ): ValidateFunction<T> {
   const ajv = new Ajv({ $data: true, allErrors: true })
   ajv_keywords(ajv)
-  ajv_errors(ajv)
+  ajv_errors(ajv, { singleError: true })
```

Comment:
Not sure what this does exactly, but doesn't it go against the `allErrors: true` that we have 2 lines above? I think it's important that all errors are reported.

Reply:
It joins errors with `;` just in very specific cases where the `instancePath` is the same